src/new_index/schema.rs

```diff
@@ -1476,31 +1514,64 @@ struct ScriptCacheKey {
 }
 
 struct StatsCacheRow {
-    key: ScriptCacheKey,
+    key: StatsCacheKey,
     value: Bytes,
 }
 
+#[derive(Serialize, Deserialize)]
+struct StatsCacheKey {
+    code: u8,
+    scripthash: FullHash,
+    color_id: StatsKey,
+}
+
 impl StatsCacheRow {
-    fn new(scripthash: &[u8], stats: &ScriptStats, blockhash: &BlockHash) -> Self {
+    fn new(
+        scripthash: &[u8],
+        color_id: StatsKey,
+        stats: &ScriptStats,
+        blockhash: &BlockHash,
+    ) -> Self {
         StatsCacheRow {
-            key: ScriptCacheKey {
+            key: StatsCacheKey {
                 code: b'A',
                 scripthash: full_hash(scripthash),
+                color_id: color_id,
             },
             value: bincode::serialize(&(stats, blockhash)).unwrap(),
         }
     }
 
-    pub fn key(scripthash: &[u8]) -> Bytes {
+    pub fn filter(scripthash: &[u8]) -> Bytes {
```

Comment:
It seems that it is just generating prefix data. Isn't it a better name to leave `key`?